### ISSUE TITLE
Add ability to clear batch and review whether batch.execute() should clear the batch so it can be reused

### ISSUE THREAD

**Issue by stefanrybacki** [jOOQ/jOOQ#14104]
### Use case

I was using `batch(dsl.insertInto(...))` with custom chunked calls to `batch.bind(...)` and wanted to call `batch.execute()` after each chunk. However, it seems batch is not cleared after call to `execute`, so I ended up with data integrity errors regarding duplicate entries as previously batched bound entries where submitted with the second call (2nd chunk) to `execute`.

### Possible solution

Either clear batch bound entries after `execute` was called or at least add option to clear already bound values from batch.

### Possible workarounds

recreate batch object after each chunk.

### jOOQ Version

3.16.10

### Database product and version

SQLite (for testing jooq)

### Java Version

11

**Comment by lukaseder** [comment]
Thanks a lot for your feature request. Changing existing behaviour is certainly not an option, so `execute` can't clear the batch on its own, much like any other `Query::execute` call won't clear the bind values, or do any other sort of "clearing". A `Query` is a command model, and all the contents of the command are sent to the server upon calling `execute()`. Calling it twice shouldn't change anything in the command, it would be quite unexpected.

This leaves us with the option of `clear()`. As we're continuously moving towards more immutability throughout our API, I wonder why you wouldn't want to simply create a new batch, rather than reusing the existing one?

**Comment by stefanrybacki** [comment]
I agree a clear method would suffice. The reason I did not create a new batch object initially was:

1. I expected execute to clear the batch
2. I am in a lambda closure while iterating chunks of data, but the batch was created outside of the lambda closure
3. I assumed creating a new batch might decrease performance

**Comment by lukaseder** [comment]
> 2\. I am in a lambda closure while iterating chunks of data, but the batch was created outside of the lambda closure

Does the loader API help with that, perhaps? https://www.jooq.org/doc/latest/manual/sql-execution/importing/importing-api/

> 3\. I assumed creating a new batch might decrease performance

The main difference will be that the SQL string is re-generated. It would still be re-generated if we add `clear()` to the `Batch`, at first. But in the future, we might be able to avoid that re-generation.

But in any ordinary application, that shouldn't matter...

I'll postpone the implementation for now (if anything is implemented at all) to collect more user feedback.
